### PR TITLE
SONAR-10031 Stop computing avatar hash in web app

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-server/src/test/java/org/sonar/server/user/ws/CurrentActionTest.java [hunk 1]
```diff
@@ -91,9 +92,9 @@ public void return_minimal_user_info() {
     CurrentWsResponse response = call();
 
     assertThat(response)
-      .extracting(CurrentWsResponse::getIsLoggedIn, CurrentWsResponse::getLogin, CurrentWsResponse::getName, CurrentWsResponse::getLocal,
+      .extracting(CurrentWsResponse::getIsLoggedIn, CurrentWsResponse::getLogin, CurrentWsResponse::getName, CurrentWsResponse::getAvatar, CurrentWsResponse::getLocal,
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/user/ws/CurrentActionTest.java]: Please do not check `CurrentWsResponse::getAvatar` but `CurrentWsResponse::hasAvatar` and expect the value to be `false`

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java [hunk 2]
```diff
@@ -95,6 +98,7 @@ private CurrentWsResponse toWsResponse(DbSession dbSession, String userLogin) {
       .setPermissions(Permissions.newBuilder().addAllGlobal(getGlobalPermissions()).build())
       .setShowOnboardingTutorial(!user.isOnboarded());
     setNullable(emptyToNull(user.getEmail()), builder::setEmail);
+    setNullable(emptyToNull(user.getEmail()), u -> builder.setAvatar(avatarResolver.create(user)));
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java]: Do not forget to update example https://github.com/SonarSource/sonarqube/blob/master/server/sonar-server/src/main/resources/org/sonar/server/user/ws/current-example.json#L5

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java [hunk 3]
```diff
@@ -95,6 +98,7 @@ private CurrentWsResponse toWsResponse(DbSession dbSession, String userLogin) {
       .setPermissions(Permissions.newBuilder().addAllGlobal(getGlobalPermissions()).build())
       .setShowOnboardingTutorial(!user.isOnboarded());
     setNullable(emptyToNull(user.getEmail()), builder::setEmail);
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java]: Existing email fields should be deprecated => Please update the JIRA ticket to say 'email' field is deprecated and add a changelog entry (line 65)

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java [hunk 4]
```diff
@@ -149,6 +153,7 @@ private static UsersWsResponse buildResponse(List<UserDto> users, List<UserPermi
         .setLogin(user.getLogin())
         .addAllPermissions(permissionsByUserId.get(user.getId()));
       setNullable(user.getEmail(), userResponse::setEmail);
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java]: Existing email fields should be deprecated => Please update the JIRA ticket to say 'email' field is deprecated and add a changelog entry

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/TemplateUsersAction.java [hunk 5]
```diff
@@ -133,6 +137,7 @@ private static PermissionQuery buildQuery(Request wsRequest, PermissionTemplateD
         .addAllPermissions(permissionsByUserId.get(user.getId()));
       setNullable(user.getEmail(), userResponse::setEmail);
       setNullable(user.getName(), userResponse::setName);
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/TemplateUsersAction.java]: Existing email fields should be deprecated => Please update the JIRA ticket to say 'email' field is deprecated and add a changelog entry

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java (hunk 4 above)
Comment (stas-vilchik) [reply server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java]: No, emails are still used and displayed on the page.

Comment (julienlancelot) [reply server/sonar-server/src/main/java/org/sonar/server/permission/ws/UsersAction.java]: ohhhh :(

**Pull request merged**